zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -177,10 +214,15 @@ public void deleteProcess(final ProcessRecord processRecord) {
     processDefinitionKey.wrapLong(processRecord.getProcessDefinitionKey());
     processId.wrapString(processRecord.getBpmnProcessId());
     processVersion.wrapLong(processRecord.getVersion());
+    deploymentKey.wrapLong(processRecord.getDeploymentKey());
 
     processColumnFamily.deleteExisting(tenantAwareProcessDefinitionKey);
     processByIdAndVersionColumnFamily.deleteExisting(tenantAwareProcessIdAndVersionKey);
 
+    // use deleteIfExists as no entry exists for older records that do not have a deployment key yet
+    processDefinitionKeyByProcessIdAndDeploymentKeyColumnFamily.deleteIfExists(
+        tenantAwareProcessIdAndDeploymentKey);
```

Comment:
💭 I was a bit unsure whether I should rather add a new version of the `ProcessDeletedApplier` and only remove the entry from this column family in that new version (similar to storing the entry from `ProcessCreatedV2Applier`).
However, you could still delete previously deployed processes for which no deployment key was added to the process record in the first place (and no entry in this column family would exist). Using a new version of the event applier for this case wouldn't make sense I think.
One idea that came to my mind was to decide in `ResourceDeletionProcessor.deleteProcess` which version of the "process deleted" record to append depending on whether the deployment key is present in the deployed process, so that a new record version and new event applier would only be used if the deployment key is actually set.

Reply:
>decide in ResourceDeletionProcessor.deleteProcess which version of the "process deleted" record to append depending on whether the deployment key is present in the deployed process.

This feels a bit too dynamic for me, and abuses the record version to achieve different logic. I'd like to avoid this.

Reply:
Perfect, thanks 👍🏻